app/util/decorators.py

```diff
@@ -27,6 +27,47 @@ def function_with_retries(*args, **kwargs):
     return method_decorator
 
 
+def retry_on_exception_exponential_backoff(exceptions, initial_delay=0.1, max_delay=10):
```

Comment:
Another option would be to allow specification of the _total_ amount of delay (the sum of all the delays), and optionally the exponential scale factor, then we could express the linear retry decorator as a special case of the exponential one. I'm fine with this as you've written it though too.